app/controllers/sessions_controller.rb

```diff
@@ -1,13 +1,24 @@
 class SessionsController < ApplicationController
   def new
+
   end
 
   def create
-    if user = User.from_omniauth(request.env["omniauth.auth"])
-      session[:user_id] = user.id
-      redirect_to root_path
+    if request.env["omniauth.auth"]
+      if user = User.from_omniauth(request.env["omniauth.auth"])
+        session[:user_id] = user.id
+        redirect_to root_path
```

Comment:
this looks good, i think we can refactor this down once we get the FB oauth in there and try to tighten it all up

Reply:
That was my next step, just wanted to get a working copy up since it's a big change

Reply:
thinking something like... 

    user = oauth_info ? oauth_login(User.locate_by(oauth_info, true)) : manual_login(User.locate_by(params[:session][:email]))

Reply:
I'm not sure where you're going with this but I'll leave it for now and we can refactor this together when we get to facebook oauth.